include/clangmetatool/tool_application_support.h

```diff
@@ -19,11 +19,18 @@ struct ToolApplicationSupport {
    * clang is installed, and that this clang-based application has been
    * installed along with the required resource files. If the installation is
    * broken, report a fatal error and exit.
+   *
+   * \param compilations Compilation Database to use for this run
+   * \param sourcePathList List of files to process during this run
+   * \param invokedArgv0 Path to the invoked clangmetatool based executable
+   *                     This must be as it appears in 'argv[0]'
+   * \param mainAddr The address of the 'main' function from the calling
+   *                 executable
    */
   static void
   verifyInstallation(const clang::tooling::CompilationDatabase &compilations,
-                     const std::vector<std::string> &sourcePathList);
-
+                     const std::vector<std::string> &sourcePathList,
+                     const std::string &invokedArgv0, void *mainAddr = nullptr);
```

Comment:
We can probably skip the mainAddr, i think its only used for esoteric cases. argv[0] is very useful though